src/http/request.cr

```diff
@@ -26,7 +30,9 @@ class HTTP::Request
 
   def to_io(io)
     io << @method << " " << @path << " " << @version << "\r\n"
-    HTTP.serialize_headers_and_body(io, @headers, @body)
+    cookies = @cookies
+    headers = cookies ? cookies.add_request_headers(@headers) : @headers
+    HTTP.serialize_headers_and_body(io, headers, @body)
```

Comment:
The `add_request_headers` is misleading (to me). I'd thought it would add cookies from headers to the collection, not the other way around.